go-controller/pkg/ovn/egressip.go

```diff
@@ -1496,11 +1510,11 @@ func (e *egressIPZoneController) addPodEgressIPAssignment(egressIPName string, s
 				return fmt.Errorf("unable to create NAT rule ops for status: %v, err: %v", status, err)
 			}
 		}
-		if config.OVNKubernetesFeature.EnableInterconnect && (loadedPodNode && !isLocalZonePod) {
+		if config.OVNKubernetesFeature.EnableInterconnect && !isOVNManagedNetwork && (loadedPodNode && !isLocalZonePod) {
 			// configure reroute for non-local-zone pods on egress nodes
-			ops, err = e.createStaticRouteOps(ops, podIPs, status, egressIPName, nextHopIP)
```

Comment:
@tssurya I tested this PR against https://github.com/ovn-org/ovn-kubernetes/pull/3897
Non-IC mode, the e2es works fine.
For non-IC node, it doesnt work. Theres LRPs reroutes missing. I dont see the reroutes here for EIP multi nic to reroute out the mgnt port.

Reply:
I am confident its not an e2e issue because ive been using them to test prior to this PR and they worked fine (needed disable the gw next hop cleanup + ignore the failure with pod2pod communications in certain scenarios)

Reply:
Rebuilt the cluster to add logs and it now passes. Possible race?
Both non-ovn managed eips + mixed (ovn +non-ovn managed) e2es both passes now.

Reply:
Putting this into a loop now - rebuild cluster + exec tests. Will report back.

Reply:
The logic in this PR is good anyway - any possible race is outside this PRs remit.

Reply:
thanks Martin for testing this out!
I'll address Tim's comments tomorrow and then we should be ready for merging this.

Reply:
Havent seen the flake again after 10 attempts. E2Es passed every time.

Reply:
Never saw the failure again after 50 cycles

Reply:
awesome thanks! resolving this comment!